src/app/features/tech-record/components/tech-record-generate-plate/tech-record-generate-plate.component.ts

```diff
@@ -0,0 +1,93 @@
+import { Component, OnChanges, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { GlobalErrorService } from '@core/components/global-error/global-error.service';
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { CustomFormGroup, FormNode, FormNodeOption, FormNodeTypes, FormNodeWidth } from '@forms/services/dynamic-form.types';
+import { TechRecordModel, VehicleTechRecordModel } from '@models/vehicle-tech-record.model';
+import { Store } from '@ngrx/store';
+import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+import { generatePlate, generatePlateSuccess } from '@store/technical-records';
+import { TechnicalRecordServiceState } from '@store/technical-records/reducers/technical-record-service.reducer';
+import { take } from 'rxjs';
+import { ValidatorNames } from '@forms/models/validators.enum';
+import { Actions, ofType } from '@ngrx/effects';
+import { PlatesInner } from '@api/vehicle';
+
+@Component({
+  selector: 'app-generate-plate',
+  templateUrl: './tech-record-generate-plate.component.html',
+  styleUrls: ['./tech-record-generate-plate.component.scss']
+})
+export class GeneratePlateComponent implements OnInit, OnChanges {
+  reasons: Array<FormNodeOption<string>> = [
+    { label: 'Free replacement', value: PlatesInner.PlateReasonForIssueEnum.FreeReplacement },
+    { label: 'Replacement', value: PlatesInner.PlateReasonForIssueEnum.Replacement },
+    { label: 'Destroyed', value: PlatesInner.PlateReasonForIssueEnum.Destroyed },
+    { label: 'Provisional', value: PlatesInner.PlateReasonForIssueEnum.Provisional },
+    { label: 'Original', value: PlatesInner.PlateReasonForIssueEnum.Original },
+    { label: 'Manual', value: PlatesInner.PlateReasonForIssueEnum.Manual }
+  ];
+
+  vehicle?: VehicleTechRecordModel;
+  currentTechRecord?: TechRecordModel;
+  form: CustomFormGroup;
+  width: FormNodeWidth = FormNodeWidth.L;
+
+  template: FormNode = {
```

Comment:
The same exact batch of comments for `app-generate-letter` also apply for this component.